prow/plugins/label/label.go

```diff
@@ -228,12 +242,29 @@ func handleComment(gc githubClient, log *logrus.Entry, config plugins.Label, e *
 			gc.CreateComment(org, repo, e.Number, plugins.FormatResponseRaw(bodyWithoutComments, e.HTMLURL, e.User.Login, canNotSetLabelReason))
 			continue
 		}
-
+		uniqueLabelsMap := getExistingUniquePrefixLabels(config, labels)
+		var labelsWithPrefixExist = false
+		for prefix := range uniqueLabelsMap {
+			if matched := strings.HasPrefix(labelToAdd, prefix); matched {
+				labelPrefix = prefix
+				labelsWithPrefixExist = true
+			}
+		}
+		// if the labelToAdd matches a unique prefix, remove all pre existing
+		// labels that match this unique prefix, and then add the new label
+		if labelsWithPrefixExist {
+			for _, label := range uniqueLabelsMap[labelPrefix] {
+				if label != labelToAdd && github.HasLabel(label, labels) {
+					if err := gc.RemoveLabel(org, repo, e.Number, label); err != nil {
```

Comment:
Ability to remote a labelj can be behind a permission (see L279), so we need to test whether the user is able to remove the label (and handle that situation well, as ignoring that condition would lead to a duplicate label on the issue/pr)

Reply:
Hi @petr-muller  thanks for reviewing this!
I wanted some clarification on this comment.

[ L235](https://github.com/kubernetes/test-infra/blob/b8337a79fb2afba511d78f8b7575ab62d145c49d/prow/plugins/label/label.go#L227-L230) already has the check which checks user permissions (which when fails, does not enter the de-duplication logic that I have added, and you have pointed to).
 
 I'm not sure how this can lead to duplicate labels?

Reply:
L235 checks for the permission to add the new label, but your L258 removes a **different** label, which the user may not have a permission to remove. Imagine this:

- Prow is configured to restrict `priority/urgent` to senior engineers
- Prow is configured to only allow one label with `priority/` prefix
- PR has `priority/urgent`
- Non-senior engineer labels the PR with `priority/low`

Your code would remove the `priority/urgent` (which the user has no permission to remove normally). If you fix this simply by denying the removal but allow adding `priority/low` then you have duplicate labels.